oci/layout/oci_dest_test.go

```diff
@@ -59,3 +60,59 @@ func TestPutBlobDigestFailure(t *testing.T) {
 	require.Error(t, err)
 	require.True(t, os.IsNotExist(err))
 }
+
+// TestPutManifestAppendsToExistingManifest tests that new manifests are getting added to existing index.
+func TestPutManifestAppendsToExistingManifest(t *testing.T) {
+	ref, tmpDir := refToTempOCI(t)
+	defer os.RemoveAll(tmpDir)
+
+	ociRef, ok := ref.(ociReference)
+	require.True(t, ok)
+
+	index, err := ociRef.getIndex()
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(index.Manifests), "Unexpected number of manifests")
+
+	putTestManifest(t, ociRef, tmpDir)
+
+	index, err = ociRef.getIndex()
+	assert.NoError(t, err)
+	assert.Equal(t, 2, len(index.Manifests), "Unexpected number of manifests")
+}
+
+// TestPutManifestTwice tests that existing manifest gets updated and not appended.
+func TestPutManifestTwice(t *testing.T) {
+	ref, tmpDir := refToTempOCI(t)
+	defer os.RemoveAll(tmpDir)
+
+	ociRef, ok := ref.(ociReference)
+	require.True(t, ok)
+
+	putTestManifest(t, ociRef, tmpDir)
+	putTestManifest(t, ociRef, tmpDir)
+
+	index, err := ociRef.getIndex()
+	assert.NoError(t, err)
+	assert.Equal(t, 2, len(index.Manifests), "Unexpected number of manifests")
+}
+
+func putTestManifest(t *testing.T, ociRef ociReference, tmpDir string) {
+	imageDest, err := newImageDestination(ociRef)
+	assert.NoError(t, err)
+
+	data := []byte("abc")
+	err = imageDest.PutManifest(data)
+	assert.NoError(t, err)
+
+	err = imageDest.Commit()
+	assert.NoError(t, err)
+
+	paths := []string{}
+	filepath.Walk(tmpDir, func(path string, info os.FileInfo, err error) error {
+		paths = append(paths, path)
+		return nil
+	})
+
+	digest := "ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad"
+	assert.Contains(t, paths, filepath.Join(tmpDir, "blobs", "sha256", digest), "The OCI directory does not contain the new manifest data")
```

Comment:
(if this is checking for a single path with a known name, the code could just as well use the `exists` function and check for that specific name. But this is only a test, and it works, so whatever.)